### PR TITLE
Increase E2E timeouts

### PR DESCRIPTION
**Pull request by tamirkamara**
## What is being addressed

E2Es are taking more time to run and reach times in the tests level and in the workflow level.

## How is this addressed

- Increase timeouts for shared services E2Es
- Increase timeout for the E2E job
- Allow the branch workflow to receive a parameter for how many tests to run in parallel (default is still 1)
- [small] update devcontainer definition

### CODE REVIEW

File: .github/workflows/deploy_tre.yml [hunk 1]
```diff
@@ -5,8 +5,8 @@ name: Deploy Azure TRE
 
 on:  # yamllint disable-line rule:truthy
   schedule:
-    # 1am each night https://crontab.guru/#0_1_*_*_*
-    - cron: "0 1 * * *"
+    # 1am each night https://crontab.guru/#0_0_*_*_*
+    - cron: "0 0 * * *"
```
Comment (guybartal) [inline .github/workflows/deploy_tre.yml:9]: Did you disable nightly build on purpose?

**Review by guybartal — APPROVED** [review]
LGTM, just make sure the nightly build was disabled intentionally.

File: .github/workflows/deploy_tre.yml (hunk 1 above)
Comment (tamirkamara) [reply .github/workflows/deploy_tre.yml:9]: No. What makes you think that?

Comment (guybartal) [reply .github/workflows/deploy_tre.yml:9]: Sorry, my mistake.

Comment (guybartal) [reply .github/workflows/deploy_tre.yml:9]: just need to update the 1am to midnight in the comment.

**Comment by tamirkamara** [comment]
/test-force-approved
changes were validated on the branch run.

**Pull request merged**